pkg/manager/internal.go

```diff
@@ -68,8 +70,14 @@ type controllerManager struct {
 	errChan              chan error
 	runnables            *runnables
 
-	// cluster holds a variety of methods to interact with a cluster. Required.
-	cluster cluster.Cluster
+	// defaultCluster holds a variety of methods to interact with a defaultCluster. Required.
+	defaultCluster        cluster.Cluster
+	defaultClusterOptions cluster.Option
+
+	// engagedCluster is a map of engaged clusters. The can come and go as the manager is running.
+	engagedClustersLock   sync.RWMutex
+	engagedClusters       map[string]cluster.Cluster
```

Comment:
It would be super helpful if we can expose the set of engagedClusters from the manager. In [Kueue](https://github.com/kubernetes-sigs/kueue/tree/main) for a multi cluster implementation, we have a list of [active and missing clusters](https://github.com/kubernetes-sigs/kueue/blob/404a0cd659b2331b232cf5ddf93a1b67d5cc146c/pkg/controller/admissionchecks/multikueue/admissioncheck.go#L86-L87) based on available quota for admitting workloads.

Reply:
For my understanding, you'd be looking for a list of the keys for that map, is that correct (i.e. the cluster identifiers, not the `cluster.Cluster` objects directly)?

Reply:
Yes, a unique identifier for the cluster, so that we can fetch the relevant client and submit workloads in that cluster.